astropy/cosmology/io/tests/test_latex.py

```diff
@@ -0,0 +1,83 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+# THIRD PARTY
+import pytest
+
+# LOCAL
+from astropy.cosmology.io.latex import _FORMAT_TABLE, write_latex
+from astropy.table import QTable, Table
+
+from .base import ReadWriteDirectTestBase, ReadWriteTestMixinBase
+
+class WriteLATEXTestMixin(ReadWriteTestMixinBase):
+    """
+    Tests for a Cosmology[Write] with ``format="latex"``.
+    This class will not be directly called by :mod:`pytest` since its name does
+    not begin with ``Test``. To activate the contained tests this class must
+    be inherited in a subclass. Subclasses must dfine a :func:`pytest.fixture`
+    ``cosmo`` that returns/yields an instance of a |Cosmology|.
+    See ``TestCosmology`` for an example.
+    """
+     
+    def test_to_latex_failed_cls(self, write, tmp_path):
+        """Test failed table type."""
+        fp = tmp_path / "test_to_latex_failed_cls.tex"
+
+        with pytest.raises(TypeError, match="'cls' must be"):
+            write(fp, format="latex", cls=list)
+
+    @pytest.mark.parametrize("tbl_cls", [QTable, Table])
+    def test_to_latex_cls(self, write, tbl_cls, tmp_path):
+        fp = tmp_path / "test_to_latex_cls.tex"
+        write(fp, format="latex", cls=tbl_cls)
+    
+    def test_latex_columns(self, write, tmp_path):
+        fp = tmp_path / "test_rename_latex_columns.tex"
+        write(fp, format="latex", latex_names=True)
+        tbl = QTable.read(fp)
+        # asserts each column name has not been reverted yet
+        # For now, Cosmology class and name are stored in first 2 slots
+        for column_name in tbl.colnames[2:]:
+            assert column_name in _FORMAT_TABLE.values()
+    
+    def test_write_latex_invalid_format(self, write, tmp_path):
+        fp = tmp_path / "test_write_latex_invalid_format.tex"
+        # Test using non-Latex format
+        with pytest.raises(ValueError, match="format must be 'latex'"):
```

Comment:
This test is failing because file type suffixes don't need to match the format of the contained data. It's poor form but technically allowed to save html to a `.tex`.